dali/image/tiff.cc

```diff
@@ -0,0 +1,111 @@
+#include "tiff.h"
+
+namespace dali {
+
+namespace {
+
+const std::vector<int> header_intel = {77, 77, 0, 42};
+const std::vector<int> header_motorola = {73, 73, 42, 0};
+const int COUNT_SIZE = 2;
+const int ENTRY_SIZE = 12;
+const int WIDTH_TAG = 256;
+const int HEIGHT_TAG = 257;
+
+
+cv::Mat DecodeTiff(const unsigned char *tiff, int size, DALIImageType image_type) {
+    std::vector<char> buf(tiff, tiff + size);
+    return cv::imdecode(buf, image_type == DALI_GRAY ? 0 : 1);
+}
+
+
+bool check_header(const unsigned char *tiff, const std::vector<int> &header) {
+    for (unsigned int i = 0; i < header.size(); i++) {
+        if (tiff[i] != header[i]) {
+            return false;
+        }
+    }
+    return true;
+}
+
+
+bool is_little_endian(const unsigned char *tiff) {
+    return check_header(tiff, header_intel);
+}
+
+} // namespace
+
+bool CheckIsTiff(const unsigned char *tiff) {
+    assert(tiff);
+    return check_header(tiff, header_intel) || check_header(tiff, header_motorola);
+}
+
+
+DALIError_t GetTiffImageDims(const unsigned char *tiff, int size, int *h, int *w) {
+    assert(CheckIsTiff(tiff));
```

Comment:
I think we don't need that check here.

Reply:
How about asserting tiff, h and w pointers?

Reply:
It's intended to be development-time check for future calls.
`assert()` macro does nothing (e.g. is expanded to a blank line), whenever `NDEBUG` flag is defined. Cmake defines this flag whenever Release configuration is built. Therefore, this check is performed only during Debug build and serves and development check, e.g. when somebody decides to call `GetTiffImageDims` not on a TIFF, but on JPEG for example.
I can imagine, that somebody could do such call, so implied bug could be noticed earlier

Reply:
Yes, tiff h and w shall be asserted also. I'll add it